pkg/services/datasources/service/datasource_service.go

```diff
@@ -321,6 +327,11 @@ func (s *Service) DecryptedValues(ctx context.Context, ds *datasources.DataSourc
 	return decryptedValues, nil
 }
 
+func shouldIgnoreDecryptedValue(ds *datasources.DataSource) bool {
+	// special case for ---Grafana-- datasource, we just need to return an empty decrypted value
+	return ds.Uid == grafanads.DatasourceUID && ds.Name == grafanads.DatasourceName
```

Comment:
I understand that the Grafana DS is a bit of a snowflake, but could you highlight why the Grafana DS is a special case for secrets in particular?

Reply:
I will update the comment, but the problem is that the grafana ds does not have a secret/password. In our current implementation, if we look for a password but could not find it in the database, it logs an error. This was reported by someone checking the ops instance. As the grafana ds is added in runtime (not provisioned) we could not add a default password, so we added this logic to avoid asking for the password so we don't log the error

Reply:
Is there a problem with not having a secret? Would this happen for other datasources?

Reply:
Not really a problem, but for all datasources it looks for the secret by default AFAIK. In the case of the Grafana datasrouce, it works ok without it, but it logs an error that could be taken as a false error

Reply:
We are not sure that there are other datasources without secret/pass, at least we could not find any other

Reply:
So this can theoretically occur for any datasource which does not have a secret? What about datasources for community plugins? I'm not should that should be considered an error 🤔

Reply:
I feel like it will likely lead to more problems of maintaining a special list for these cases - I'm not even sure it warrants a warn log. If someone wants to create a datasource plugin that requires no secrets (only accesses public information for example), then I think this is a valid use case. Is there a risk with no log (or just switching it to `debug`)?

Reply:
> Is there a risk with no log (or just switching it to debug)?

Not really, AFAIK. 
We thought of keeping it as a warn to highlight it in case there is an error, and someone is looking for a reason. 
But on the other hand, if a datasource requires a password and can't retrieve it, it will show an error when trying to make a query.
If you think would be better, I can make another PR just change it into a warn (in fact, I already had it [here](https://github.com/grafana/grafana/pull/52937) )

Reply:
Warn is still shows up in the log by default and I think can cause confusion, as it will always show up for certain data sources no matter what. I would be of the opinion to remove or change to debug!

Reply:
Ok, back to debug then, I would prefer to keep the log if that is ok
cc: @mmandrus

Reply:
replaced with this PR https://github.com/grafana/grafana/pull/53007